blocks/library/embed/index.js

```diff
@@ -156,6 +163,13 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 					</BlockControls>
 				);
 
+				if ( fetching && fromPaste ) {
```

Comment:
Why do we need to differentiate from normal fetch?

Reply:
This seems like a good candidate for a comment :)

If I pasted a URL into a paragraph, then I'd see the "Embedding..." UI from the Embed block pop up while the fetch happened. It caused a bit of UI jumping and looked out of place. With this, we see just a paragraph with the link in it while the fetch goes on, and it feels like you don't leave the paragraph block while that's happening.

Reply:
Sure, but don't we then convert to an embed block at the end anyway?

Reply:
If we can embed the URL, we do. If we can't, it becomes a paragraph block. So with this flow, the user sees "Paragraph block -> pasted URL in a paragrph -> (Embedded content | URL in paragraph block)"

Without it, user sees, "Paragraph block -> Large grey block saying "Embedding" -> (Embedded content | URL in paragraph block)" and the experience doesn't feel as smooth.

Reply:
Yep, this is assuming we don't have the paragraph fallback then.

Reply:
If we didn't have paragraph fallback, this can go completely. I'd like to tackle that separately though, because if we're going to go the whole Facebook style "guess the preview", then we need to give the user more control over what gets displayed as part of that preview, or allow them to get rid of it completely. That's quite a bit more work, and I think this is a good first step on the way to getting there.

Reply:
To expand on this, to keep oEmbed happy, pasting a URL that oEmbed cannot handle would result in a paragraph block with the URL in it, and an "Embedded URL Preview" block (or some similar name) which generates a preview separate from the link in the paragraph block. That's so the preview can be optional, and have bits edited and customised, should our guesses get it wrong.